tests/test_sampling_rates.py

```diff
@@ -254,3 +260,60 @@ def validator(data):
 
             if not all((d == decisions[0] for d in decisions)):
                 raise ValueError(f"Sampling decisions are not deterministic for trace_id {trace_id}: {decisions}")
+
+    def _load_csv_sampling_decisions(self):
+        cases = []
+        with open("tests/fixtures/sampling_rates.csv", newline="", encoding="utf-8") as csv_file:
+            r = csv.reader(csv_file)
+            for row in r:
+                if float(row[1]) != context.tracer_sampling_rate:
+                    # This test can only run on an app with a static sampling rate (default: 0.5).
+                    logger.warning(
+                        "skipped fixture trace_id:%s because of sampling rate different from %s", row[0], row[1]
+                    )
+                    continue
+                cases.append((int(row[0]), float(row[1]), int(row[2])))
+        return cases
+
+    def setup_sample_rate_function(self):
+        self.requests_expected_decision = []
+
+        for trace_id, sampling_rate, sampling_priority in self._load_csv_sampling_decisions():
+            sampling_decision = priority_should_be_kept(sampling_priority)
+            # Check that the input .csv is valid according to this test's implementation of trace_should_be_kept.
+            assert sampling_decision is trace_should_be_kept(sampling_rate, trace_id)
```

Comment:
Setup function must never fail, any assertion should be made on test method.

IIUC, it's only csv data validation. In that case, I think it would be better to move it in `_load_csv_sampling_decisions`? WDYT ?

Reply:
Yes it's ok. If ever it fails, you will have a big nasty SYSTEM TESTS INTERNAL ERROR that does not look similar to a regular failure in stdout.